rclcpp/include/rclcpp/topic_statistics/subscription_topic_statistics.hpp

```diff
@@ -163,6 +167,7 @@ class SubscriptionTopicStatistics
   /// Stop all collectors, clear measurements, stop publishing timer, and reset publisher.
   void tear_down()
   {
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
This may not be required, since the [Collector::Stop()](https://github.com/ros-tooling/libstatistics_collector/blob/master/src/libstatistics_collector/collector/collector.cpp#L39-L53) method is already guarded by a mutex?

Reply:
> This may not be required, since the [Collector::Stop()](https://github.com/ros-tooling/libstatistics_collector/blob/master/src/libstatistics_collector/collector/collector.cpp#L39-L53) method is already guarded by a mutex?

Line 175 clears `subscriber_statistics_collectors_` though, so that should at least be protected.